core/src/main/java/hudson/model/Queue.java

```diff
@@ -2489,9 +2536,14 @@ public CauseOfBlockage getCauseOfBlockage() {
                     if (nodes.size() != 1)      return new BecauseLabelIsOffline(label);
                     else                        return new BecauseNodeIsOffline(nodes.iterator().next());
                 } else {
+                    if (transientCausesOfBlockage != null) {
+                        return new CompositeCauseOfBlockage(transientCausesOfBlockage);
+                    }
```

Comment:
The main concern here is that this might mask cases where `BecauseLabelIsBusy` is the best explanation, yet there are some free executors on nonmatching nodes; in that case you will get a set of `Node.LabelMissing`s via `Node.canTake`, which is less helpful. Looking into whether `getIdleExecutors` can be used to avoid regressing that case.